src/Telemetry/Data_Providers/Debug_Data.php

```diff
@@ -46,9 +47,34 @@ public function get_data(): array {
 				unset( $plugins[ $active_plugin ] );
 			}
 		}
+
 		$info['telemetry-active-plugins']['fields']   = $active;
 		$info['telemetry-inactive-plugins']['fields'] = $plugins;
 
 		return $info;
 	}
+
+	/**
+	 * Some data in Site Health is marked as private.
+	 * This ensures we don't pass that on to the servers.
+	 *
+	 * @since TBD
+	 *
+	 * @param array<string,mixed> $data Raw Site Health data.
+	 *
+	 * @return array<string,mixed> Filtered Site Health data.
+	 */
+	public function clean_private_data( $data ): array {
+		foreach ( $data as &$details ) {
+			// remove private info.
+			$details['fields'] = array_filter(
+				$details['fields'],
+				function ( $field ) {
+					return empty( $field['private'] );
+				}
+			);
```

Comment:
When the data is removed, the `array_filter` function will leave "holes" in the numeric indexes.
Some code down the line might assume continuity in the indexes.
Use `array_values` to restore continuity:

```suggestion
			$details['fields'] = array_values( array_filter(
				$details['fields'],
				function ( $field ) {
					return empty( $field['private'] );
				}
			) );
```